### PR TITLE
cvd bug report API

### PR DESCRIPTION
**Pull request by ser-io**

### CODE REVIEW

File: frontend/src/host_orchestrator/orchestrator/controller.go [hunk 1]
```diff
@@ -76,6 +76,9 @@ func (c *Controller) AddRoutes(router *mux.Router) {
 	// reached. Be prepared to retry if the deadline was reached.
 	router.Handle("/operations/{name}/:wait",
 		httpHandler(&waitOperationHandler{c.OperationManager, c.WaitOperationDuration})).Methods("POST")
+	router.Handle("/cvdbugreports",
+		httpHandler(newCreateCVDBugReportHandler(c.Config, c.OperationManager))).Methods("POST")
+	router.Handle("/cvdbugreports/{uuid}/:download", &downloadCVDBugReportHandler{c.Config}).Methods("POST")
```
Comment (jemoreira) [inline frontend/src/host_orchestrator/orchestrator/controller.go:81]: why POST? Does it delete it from the remote host?

File: frontend/src/host_orchestrator/orchestrator/createcvdbugreportaction.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+const BugReportZipFileName = "cvd_bugreport.zip"
+
+func (a *CreateCVDBugReportAction) createBugReport(uuid string, op apiv1.Operation) {
+	result := &OperationResult{}
+	filename := filepath.Join(a.paths.CVDBugReportsDir, uuid, "cvd_bugreport.zip")
+	cmd := cvd.NewCommand(
+		a.execContext,
+		[]string{"host_bugreport", "--output=" + filename},
```
Comment (jemoreira) [inline frontend/src/host_orchestrator/orchestrator/createcvdbugreportaction.go:69]: FYI: This command will block reading from `stdin` if there are more than one group running.

File: frontend/src/host_orchestrator/orchestrator/controller.go [hunk 3]
```diff
@@ -76,6 +76,9 @@ func (c *Controller) AddRoutes(router *mux.Router) {
 	// reached. Be prepared to retry if the deadline was reached.
 	router.Handle("/operations/{name}/:wait",
 		httpHandler(&waitOperationHandler{c.OperationManager, c.WaitOperationDuration})).Methods("POST")
+	router.Handle("/cvdbugreports",
```
Comment (jemoreira) [inline frontend/src/host_orchestrator/orchestrator/controller.go:79]: Despite the (misleading) command name, host bug reports in `cvd` are taken from a group, not the entire host. There should be a way to associate the bugreport to the group it was taken from, both at creation time and later. The best way to do this is to make the bugreports a subresource of `/cvds`

File: frontend/src/host_orchestrator/orchestrator/controller.go (hunk 1 above)
Comment (jemoreira) [inline frontend/src/host_orchestrator/orchestrator/controller.go:81]: For completion, add a `DELETE /{bugreportspath}/{uuid}` endpoint.

File: frontend/src/liboperator/api/v1/messages.go [hunk 4]
```diff
@@ -227,3 +227,7 @@ type EmptyResponse struct{}
 
 type StopCVDResponse = EmptyResponse
 
+type UUID struct {
```
Comment (jemoreira) [inline frontend/src/liboperator/api/v1/messages.go:230]: A full struct to represent a uuid is overkill, just a string would be enough.

**Review by jemoreira — CHANGES_REQUESTED** [review]

File: frontend/src/host_orchestrator/orchestrator/controller.go (hunk 1 above)
Comment (ser-io) [reply frontend/src/host_orchestrator/orchestrator/controller.go]: good catch, just carried the POST, changed it to GET.

File: frontend/src/host_orchestrator/orchestrator/createcvdbugreportaction.go (hunk 2 above)
Comment (ser-io) [reply frontend/src/host_orchestrator/orchestrator/createcvdbugreportaction.go:75]: good point, made the group name required.

File: frontend/src/host_orchestrator/orchestrator/controller.go (hunk 3 above)
Comment (ser-io) [reply frontend/src/host_orchestrator/orchestrator/controller.go]: Good point,

The new API looks like:

```
#  `:bugreport` is an operation of the cvd resource which creates the `cvdbugreport` resource.
POST /cvds/{group}/:bugreport

#  then we add new endpoints handling `cvdbugreport` as a main resource.
GET /cvdbugreport/{uuid}
DELETE /cvdbugreport/{uuid}
```

File: frontend/src/liboperator/api/v1/messages.go (hunk 4 above)
Comment (ser-io) [reply frontend/src/liboperator/api/v1/messages.go]: True, was under the impression a single string was not a valid JSON value.

File: frontend/src/host_orchestrator/orchestrator/controller.go (hunk 1 above)
Comment (ser-io) [reply frontend/src/host_orchestrator/orchestrator/controller.go]: Done

**Review by jemoreira — APPROVED** [review]

**Pull request merged**